cterasdk/common/object.py

```diff
@@ -1,6 +1,136 @@
+import re
 import json
+import logging
 
 
 class Object:  # pylint: disable=too-many-instance-attributes
     def __str__(self):
         return json.dumps(self, default=lambda o: o.__dict__, indent=5)
+
+
+class Device(Object):
+
+    def __init__(self, uid, version, firmware):
+        self.namespace = 'http://www.w3.org/2001/XMLSchema-instance'
+        self.location = '../../db/resources/db.xsd'
+        self.id = uid
+        self.version = version
+        self.firmware = firmware
+
+
+def delete_attrs(obj, paths):
+    """
+    Delete attributes
+
+    :param cterasdk.common.object.Object object: The object
+    :param list[str] paths: List of attributes to remove
+
+    :returns: The modified object
+    :rtype: cterasdk.common.object.Object
+    """
+    for path in paths:
+        delete_attr(obj, path)
+
+
+def delete_attr(obj, path):
+    """
+    Delete attribute
+
+    :param cterasdk.common.object.Object object: The object
+    :param str path: Attribute path
+
+    :returns: The modified object
+    :rtype: cterasdk.common.object.Object
+    """
+    parts = re.findall('[^/]+', path)
+    parent = find_attr(obj, parts[:-1])
+    remove_attr(parent, parts[-1])
+
+    if len(parts) > 1 and isinstance(parent, Object) and not parent.__dict__:
+        grandparent = find_attr(obj, parts[:-2])
+        setattr(grandparent, parts[-2], None)
+
+
+def find_attr(obj, path):
+    """
+    Find attribute
+
+    :param cterasdk.common.object.Object object: The object
+    :param str path: A string or an array of the attribute path
+
+    :returns: The attribute, or ``None`` if not found
+    """
+    parts = None
```

Comment:
This can be a one liner:
`parts = re.findall('[^/]+', path) if isinstance(path, str) else path`